File: frontend/src/microservices/advertising/objects/UpdateAdvertisementRequest.ts
// 本文件定义 advertising 模块的 `UpdateAdvertisementRequest`，作为请求参数并提供 JSON 编解码。

export type UpdateAdvertisementRequest = {
  ownerManagerId: string
  ownerType: string
  advertisementKind?: string | null
  title: string
  subtitle: string
  description: string
  imageUrl?: string | null
  ctaLabel: string
  targetResourceType: string
  targetResourceId: string
  resourceSummaryTitle?: string | null
  landingTarget?: string | null
  placement: string
  creativeJson?: string | null
  creativeWidth?: number | null
  creativeHeight?: number | null
  priority: number
  startAt: string
  endAt: string
}

export const updateAdvertisementRequestFromJson = (json: string): UpdateAdvertisementRequest =>
  JSON.parse(json) as UpdateAdvertisementRequest

export const updateAdvertisementRequestToJson = (value: UpdateAdvertisementRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/advertising/objects/UpdateAdvertisementRequest.scala
// 本文件定义 advertising 模块的 UpdateAdvertisementRequest，作为后端与前端同名的数据模型，并提供 JSON codec。
package com.typesafe.travel.advertising.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class UpdateAdvertisementRequest(
    advertisementId: String,
    ownerManagerId: String,
    ownerType: String,
    advertisementKind: Option[String],
    title: String,
    subtitle: String,
    description: String,
    imageUrl: Option[String],
    ctaLabel: String,
    targetResourceType: String,
    targetResourceId: String,
    resourceSummaryTitle: Option[String],
    landingTarget: Option[String],
    placement: String,
    creativeJson: Option[String],
    creativeWidth: Option[Int],
    creativeHeight: Option[Int],
    priority: Int,
    startAt: String,
    endAt: String
)

object UpdateAdvertisementRequest:
  given sourceEncoder: Encoder[UpdateAdvertisementRequest] = deriveEncoder
  given sourceDecoder: Decoder[UpdateAdvertisementRequest] = deriveDecoder

